ios/Runner/AppDelegate.swift

```diff
@@ -86,9 +75,49 @@ import flutter_local_notifications
         return URL(string: "ShareMedia-\(appDomain)://dataUrl=\(sharedKey)#text")
     }
     
+    // Receive displayed notifications for iOS 10 or later devices.
     override func userNotificationCenter(_ center: UNUserNotificationCenter,
                                          willPresent notification: UNNotification,
                                          withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
-        completionHandler(.alert) // shows banner even if app is in foreground
+        print("userNotificationCenter >> willPresent >> notification: \(notification)")
+        if let notificationBadgeCount = notification.request.content.badge?.intValue, notificationBadgeCount > 0 {
+            let newBadgeCount = UIApplication.shared.applicationIconBadgeNumber + notificationBadgeCount
+            print("userNotificationCenter >> willPresent >> newBadgeCount: \(newBadgeCount)")
+            updateAppBadger(currentBadgeCount: newBadgeCount)
+        }
+        
+        completionHandler(isAppForegroundActive() ? [.alert, .badge, .sound] : [])
```

Comment:
IMO, we should show notification if app is not in Foreground?

Reply:
Oh, That's right, I probably made a mistake here, I'll correct it.